mbox: do not use backtrace if using uClibc without backtrace support

Since uClibc can be configured without support for backtrace, disable
the backtrace if we are building with a uClibc that was built without
backtrace.

This is a bit hacky, and would greatly benefit from a test in ./configure
instead, but does nicely as a quick fix for now.

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
diff -durN clamav-0.98.4.orig/libclamav/mbox.c clamav-0.98.4/libclamav/mbox.c
--- clamav-0.98.4.orig/libclamav/mbox.c	2014-05-21 17:25:05.000000000 +0200
+++ clamav-0.98.4/libclamav/mbox.c	2014-11-16 17:21:04.885383286 +0100
@@ -83,7 +83,7 @@
 #include <features.h>
 #endif
 
-#if __GLIBC__ == 2 && __GLIBC_MINOR__ >= 1
+#if __GLIBC__ == 2 && __GLIBC_MINOR__ >= 1 && !defined(__UCLIBC__) || defined(__UCLIBC_HAS_BACKTRACE__)
 #define HAVE_BACKTRACE
 #endif
 #endif
